components/tabs/tab-panel-mixin.js

```diff
@@ -65,11 +65,12 @@ export const TabPanelMixin = superclass => class extends superclass {
 	updated(changedProperties) {
 		super.updated(changedProperties);
 
-		changedProperties.forEach((oldVal, prop) => {
+		changedProperties.forEach((_, prop) => {
 			if (prop === 'labelledBy') {
 				this.setAttribute('aria-labelledby', this.labelledBy);
 			} else if (prop === 'selected') {
-				if (this.selected) {
+				// assuming if this.text that we are using the old workflow
+				if (this.selected && this.text && !this.labelledBy) {
```

Comment:
I believe we don't need to dispatch this event if using the new method, as we would be expecting `selected` to get set on the tab. Let me know if this is incorrect.

Reply:
Makes sense. I agree that with it being dispatched from the tab instead if we're planning on also having the tab be what `selected` gets set on.